src/main/java/org/terasology/ReplayTestingEnvironment.java

```diff
@@ -41,11 +47,27 @@
 import java.nio.file.Path;
 import java.nio.file.Paths;
 import java.util.List;
+import java.util.function.BooleanSupplier;
 
 /**
- * A base class for replay tests to inherit to run a replay in a thread while testing some variables.
+ * A base class for tests involving a full {@link TerasologyEngine} instance that runs a replay. View the tests in the
+ * "examples" package for simple usage examples.
+ * <p>
+ * Classes that extend this one can initialize an engine through two methods: {@link #openMainMenu()} or {@link #openReplay(String, boolean)}.
+ * The openMainMenu() method initializes a headed engine and opens the game in the main menu, while the openReplay(String, boolean)
+ * initialises an engine and right after its initialisation, opens a replay. The engine of the later can be either headed
+ * or headless.
+ * <p>
+ * It is possible to get the engine through the {@link #getHost()} method, the {@link RecordAndReplayCurrentStatus}'s
+ * status through the {@link #getRecordAndReplayStatus()} method and if the host engine was initialized through the
+ * {@link #isInitialised()} method.
+ * <p>
+ * The last protected method is {@link #waitUntil(BooleanSupplier)} which is generally used to wait for the replay to get
+ * to a certain point so something can be tested at the right moment.
```

Comment:
I'd include class usage instructions rather than method descriptions here. The javadoc should describe the workflow for creating replay test by extending from `ReplayTestingEnvironment`. In the process of describing this workflow, important utility methods (like `getHost` and `waitUntil`) can be mentioned and briefly described (if necessary). The javadoc currently does not provide any information on how to create a replay test by extending this class, which methods to override, what the overriden method bodies should contain et cetera, which in my opinion is the more valuable information.